チャット画面のviewを作成

yosuketokunaga opened the pull request:
#  WHAT
チャット画面のviewを作成

#  WHY
アプリ作成に重要な画面であり必須のため。

Commit:
Change.directly

_reset.scssの置き場所を変更しました。

Commit:
Edit.front-screen

index.html.erbとapplication.scssを編集しフロント画面を作成しました。

yosuketokunaga commented:
![2017-04-04 20 09 19](https://cloud.githubusercontent.com/assets/26739922/24654384/5619b654-1974-11e7-8209-338fdb74d4ba.png)

kenchan0503 commented on app/assets/stylesheets/application.scss:
```diff
@@ -1 +1,234 @@
+  line-height: 24px;
+  font-weight: 200;
+  text-align: center;
+  color: #FFFFFF
```
同じ文字が連続している時は#FFFというように省略可能です

kenchan0503 replied:
また、複数箇所で仕様しているので変数化しましょう

kenchan0503 commented on app/assets/stylesheets/application.scss:
```diff
@@ -1 +1,234 @@
+        &:after {
+          content: '';
+          display: block;
+          clear: both;
```
mix-inに組み込みましょう

kenchan0503 commented:
wipを外しましょう

Commit:
Edit.front-screen again1

コメントいただいた点を修正(haml,scssファイル)及び新たに_clearfix.scssを作成追加した。

KoichiOzaki commented on app/assets/stylesheets/application.scss:
```diff
@@ -1 +1,236 @@
+              display: block;
+              clear: both;
+            }
+            .chat__main--body--block--name{
```
&記法を使えるところは使いましょう。
`&--name{`
のように書けるはずです。

KoichiOzaki commented on app/assets/stylesheets/application.scss:
```diff
@@ -1 +1,236 @@
+              font-weight: 600;
+              float: left;
+            }
+            .chat__main--body--block--date{
```
ここも&記法使いましょう。

KoichiOzaki replied:
以降も同じように省略可能な所は&で省略していきましょう。

KoichiOzaki commented on app/assets/stylesheets/application.scss:
```diff
@@ -1 +1,236 @@
+    }
+    &--footer {
+      height: 90px;
+      background-color: #DDDDDD;
```
`#DDD`と書きましょう。

KoichiOzaki commented on app/assets/stylesheets/application.scss:
```diff
@@ -1 +1,236 @@
+        padding-top:20px;
+        padding-bottom:20px;
+        padding-left:40px;
+        padding-right: 40px;
```
4方向のpaddingを設定するのであれば、`padding: Opx Opx Opx Opx`のように書いた方がシンプルかと思います。

Commit:
Edit.frontscreen again2

コメントいただいた&での書き換え及びpaddingでの書き方への修正見落としていたclearfix記述の追加。

madana-i commented on app/assets/stylesheets/application.scss:
```diff
@@ -1 +1,225 @@
 @import "./reset";
+@import "./clearfix";
+@import "font-awesome";
+
```
application.scssはimportを行うだけのファイルにしましょう！

madana-i commented on app/views/messages/index.html.haml:
```diff
@@ -1 +1,44 @@
+  %div.chat__side
+    %div.chat__side--user
+      %h5 User Name
+      %a.edit-your-account
```
クラス名は名称なので、動詞を用いることはなるべく避けましょう！
この場合ですと、account-edit-buttonといったような命名がよろしいかと思います！

Commit:
Edit frontsteen again3

buuttonの名称変更。
application.scssに記述していた内容をmessages.scssに移行。application.scssはimportのみ行うファ
イルとする。

miyagit commented on app/views/messages/index.html.haml:
```diff
@@ -1 +1,44 @@
+        %h2 sampleグループ
+        %p Members: yosuketokunaga
+      %div.chat__main--header__right
+        %a Edit
```
aタグではなく、link_toを使うようにしましょう。

Commit:
Edit.frontscreen again3

aタグで記述していたEditの箇所を修正致しました。

FumiyaMori commented on app/assets/stylesheets/_messages.scss:
```diff
@@ -0,0 +1,224 @@
+$font-color-white: #FFFFFF;
+$font-color-gray: #434A54;
+$font-color-blue: #38AEF0;
+$font-color-for-date:#999999;
```
変数用のファイルを別に作り、そこに変数を書くようにしましょう

FumiyaMori commented on app/assets/stylesheets/_messages.scss:
```diff
@@ -0,0 +1,224 @@
+    &--header {
+      padding-right: 40px;
+      padding-left: 40px;
+      border-bottom: 1px solid #eeeeee;
```
`#eeeeee`のような場合は`#eee`のように省略することができるので短く書いてみましょう
他にも似たような箇所があるので合わせて修正お願いします

FumiyaMori commented on app/assets/stylesheets/application.scss:
```diff
@@ -1 +1,4 @@
 @import "./reset";
+@import "./clearfix";
+@import "font-awesome";
```
外部のライブラリは最初に読み込むのが良いですね

FumiyaMori commented on app/views/messages/index.html.haml:
```diff
@@ -1 +1,44 @@
+        %textarea.chat__main--footer--block__left{placeholder: "type a message"}
+        %label.chat__main--footer--image
+          %i.fa.fa-image
+        .chat__main--footer--block__right Send
```
inputを使ってsendボタンを作りましょう

FumiyaMori commented on app/views/messages/index.html.haml:
```diff
@@ -1 +1,44 @@
+            .chat__main--body--block--date 17/04/04 16:11:11
+          %p.chat__main--body--message テスト2
+    %div.chat__main--footer
+      .chat__main--footer--block
```
formタグを使ってフォーム自体を作っておきましょう

FumiyaMori commented on app/views/messages/index.html.haml:
```diff
@@ -1 +1,44 @@
+      .chat__main--footer--block
+        %textarea.chat__main--footer--block__left{placeholder: "type a message"}
+        %label.chat__main--footer--image
+          %i.fa.fa-image
```
ファイルのアップにもinputタグを使いますので、そのように書きましょう

FumiyaMori commented on app/views/messages/index.html.haml:
```diff
@@ -1 +1,44 @@
-%p  Hello World!
+%p.notice チャットグループが作成されました。
+
+%div.chat
```
hamlではdivは省略することができます
`%div.chat`は`.chat`のように書くことができます

Commit:
Edit.frontscreen again4

変数用のファイル作成、
省略可能な色指定記述、
divの省略、
formタグの導入、
ファイルアップ用にinputタグ導入とコメントいただいたが、fontawesomeがうまく機能しないのでbuttonタグにて記述、
inputタグでsendボタンの作成

snowhork commented on app/views/messages/index.html.haml:
```diff
@@ -1 +1,54 @@
+
+          -#
+          -# %textarea.chat__main--footer--block__left{placeholder: "type a message"}
+          -# %label.chat__main--footer--image
```
こちらコメントアウトされてますが、何か意図はありますか？

不要であれば削除、意図があれば、`TODO: メッセージ投稿の時に実装する` のようにコメントを添えておくべきですね！

原則コードのコメントアウトは無い方が良いですね！

yosuketokunaga replied:
すみません。消し忘れです。修正いたします!

Commit:
Edit.frontscreen again5

コメントアウトで記述していた箇所を削除しました。
